certificate-service/app/src/main/java/se/inera/intyg/certificateservice/application/certificate/service/UpdateCertificateService.java

```diff
@@ -0,0 +1,61 @@
+package se.inera.intyg.certificateservice.application.certificate.service;
+
+import java.util.Map.Entry;
+import java.util.function.Predicate;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import se.inera.intyg.certificateservice.application.certificate.dto.CertificateDataElement;
+import se.inera.intyg.certificateservice.application.certificate.dto.UpdateCertificateRequest;
+import se.inera.intyg.certificateservice.application.certificate.dto.UpdateCertificateResponse;
+import se.inera.intyg.certificateservice.application.certificate.dto.config.CertificateDataConfigTypes;
+import se.inera.intyg.certificateservice.application.certificate.service.validation.UpdateCertificateRequestValidator;
+import se.inera.intyg.certificateservice.application.common.ActionEvaluationFactory;
+import se.inera.intyg.certificateservice.domain.certificate.model.CertificateId;
+import se.inera.intyg.certificateservice.domain.certificate.service.UpdateCertificateDomainService;
+
+@Service
+@RequiredArgsConstructor
+public class UpdateCertificateService {
+
+  private final UpdateCertificateRequestValidator updateCertificateRequestValidator;
+  private final UpdateCertificateDomainService updateCertificateDomainService;
+  private final ElementDataConverter elementDataConverter;
+  private final ActionEvaluationFactory actionEvaluationFactory;
+  private final CertificateConverter certificateConverter;
+
+  public UpdateCertificateResponse update(UpdateCertificateRequest updateCertificateRequest,
+      String certificateId) {
+    updateCertificateRequestValidator.validate(updateCertificateRequest, certificateId);
+
+    final var actionEvaluation = actionEvaluationFactory.create(
+        updateCertificateRequest.getPatient(),
+        updateCertificateRequest.getUser(),
+        updateCertificateRequest.getUnit(),
+        updateCertificateRequest.getCareUnit(),
+        updateCertificateRequest.getCareProvider()
+    );
+
+    final var elementDataList = updateCertificateRequest.getCertificate().getData()
+        .entrySet()
+        .stream()
+        .filter(removeCategories())
+        .map(entry -> elementDataConverter.convert(entry.getKey(), entry.getValue()))
+        .toList();
+
+    final var updatedCertificate = updateCertificateDomainService.update(
+        new CertificateId(certificateId),
+        elementDataList,
+        actionEvaluation
+    );
+
+    return UpdateCertificateResponse.builder()
+        .certificate(certificateConverter.convert(updatedCertificate))
+        .build();
```

Comment:
Missed it before, but you should add some resource links as well. Should work by just adding the same code as in GetCertificateService

Reply:
I think we made a misstake with converting the ResourceLinks... the links attribute sits inside the CertificateDTO and not on the same level as the "certificate".

Reply:
I been looking into this a little deeper and we have an issue actually.. We will need do make some changes to Webcert (and Common) in order to make this work correctly.

Reply:
I see, I will abort the attempt to update links and maybe we can sync the required changes tomorrow morning.